projects/kit/components/input-password/input-password.directive.ts

```diff
@@ -23,12 +30,14 @@ export class TuiInputPasswordDirective
         this.input = input;
     }
 
-    public ngDoCheck(): void {
-        if (
-            this.host.nativeFocusableElement &&
-            tuiIsInput(this.host.nativeFocusableElement)
-        ) {
-            this.host.nativeFocusableElement.type = this.host.inputType;
-        }
+    public ngAfterViewInit(): void {
+        this.isPasswordHidden$.subscribe(() => {
+            if (
+                this.host.nativeFocusableElement &&
+                tuiIsInput(this.host.nativeFocusableElement)
```

Comment:
```ts
this.isPasswordHidden$.pipe(
  filter(() => this.host.nativeFocusableElement && tuiIsInput(this.host.nativeFocusableElement))
).subscribe(() => {});
```

Reply:
unfortunately the filter pipe won't give the expected result since Typescript won't consider `this.host.nativeFocusableElement` as an `HTMLInputElement` and I'll have to use `if` inside the callback anyway